### PR TITLE
docs: Explain how to obtain the aggregate result of all iterations of a loop

### LINKED ISSUE #10775: Retrieving aggregate results from step with 'withParam' or 'withItems' causes a JSON parse error
### Pre-requisites

- [X] I have double-checked my configuration
- [ ] I can confirm the issues exists when I tested with `:latest`
- [x] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

A simple workflow causes the error `invalid character ':' after top-level value`.
I would expect to get a JSON array whose elements correspond to each execution generated by `withParam`.

Perhaps there's some problem in the [processAggregateNodeOutputs](https://github.com/argoproj/argo-workflows/blob/d307c80e7973959e54b495c7e4015bccf383385a/workflow/controller/operator.go#L2888-L2945) function?

### Version

latest

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: loop-test
  namespace: argo-workflows
  annotations:
    workflows.argoproj.io/tags: argocd
    workflows.argoproj.io/version: '>= 2.9.0'
spec:
  entrypoint: main
  templates:
  - name: main
    steps:
    - - name: execute-parallel-steps
        template: print-time-with-index
        arguments:
          parameters:
          - name: index
            value: '{{item}}'
        withParam: '["1", "2", "3"]'
    - - name: call-access-aggregate-output
        template: access-aggregate-output
        arguments:
          parameters:
          - name: aggregate-results
            value: '{{steps.execute-parallel-steps.outputs.result}}'
  - name: print-time-with-index
    inputs:
      parameters:
      - name: index
    script:
      image: alpine:latest
      command: [sh]
      source: |
        echo "{{inputs.parameters.index}}: $(date)"
  - name: access-aggregate-output
    inputs:
      parameters:
      - name: aggregate-results
        value: 'no-value'
    script:
      image: alpine:latest
      command: [sh]
      source: |
        echo 'inputs.parameters.aggregate-results: "{{inputs.parameters.aggregate-results}}"'
```

### Logs from the workflow controller

```text
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}

kubectl logs -n argo argo-workflows-workflow-controller-85d5658c9d-bjd9p | grep $workflow

time="2023-03-29T01:18:35.935Z" level=info msg="Processing workflow" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:35.952Z" level=info msg="Updated phase  -> Running" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:35.953Z" level=info msg="Steps node loop-test-g926s initialized Running" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:35.953Z" level=info msg="StepGroup node loop-test-g926s-3122930240 initialized Running" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:35.953Z" level=info msg="Pod node loop-test-g926s-180455059 initialized Pending" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:35.998Z" level=info msg="Created pod: loop-test-g926s[0].execute-parallel-steps(0:1) (loop-test-g926s-180455059)" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:35.998Z" level=info msg="Pod node loop-test-g926s-407593233 initialized Pending" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:36.043Z" level=info msg="Created pod: loop-test-g926s[0].execute-parallel-steps(1:2) (loop-test-g926s-407593233)" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:36.043Z" level=info msg="Pod node loop-test-g926s-2537052995 initialized Pending" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:36.073Z" level=info msg="Created pod: loop-test-g926s[0].execute-parallel-steps(2:3) (loop-test-g926s-2537052995)" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:36.073Z" level=info msg="Workflow step group node loop-test-g926s-3122930240 not yet completed" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:36.073Z" level=info msg="TaskSet Reconciliation" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:36.073Z" level=info msg=reconcileAgentPod namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:36.089Z" level=info msg="Workflow update successful" namespace=argo-workflows phase=Running resourceVersion=742464014 workflow=loop-test-g926s
time="2023-03-29T01:18:46.005Z" level=info msg="Processing workflow" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.005Z" level=info msg="Task-result reconciliation" namespace=argo-workflows numObjs=3 workflow=loop-test-g926s
time="2023-03-29T01:18:46.005Z" level=info msg="task-result changed" namespace=argo-workflows nodeID=loop-test-g926s-2537052995 workflow=loop-test-g926s
time="2023-03-29T01:18:46.005Z" level=info msg="task-result changed" namespace=argo-workflows nodeID=loop-test-g926s-407593233 workflow=loop-test-g926s
time="2023-03-29T01:18:46.005Z" level=info msg="task-result changed" namespace=argo-workflows nodeID=loop-test-g926s-180455059 workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=info msg="node changed" new.message= new.phase=Succeeded new.progress=0/1 nodeID=loop-test-g926s-2537052995 old.message= old.phase=Pending old.progress=0/1
time="2023-03-29T01:18:46.006Z" level=info msg="node changed" new.message= new.phase=Succeeded new.progress=0/1 nodeID=loop-test-g926s-180455059 old.message= old.phase=Pending old.progress=0/1
time="2023-03-29T01:18:46.006Z" level=info msg="node changed" new.message= new.phase=Succeeded new.progress=0/1 nodeID=loop-test-g926s-407593233 old.message= old.phase=Pending old.progress=0/1
time="2023-03-29T01:18:46.006Z" level=info msg="Step group node loop-test-g926s-3122930240 successful" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=info msg="node loop-test-g926s-3122930240 phase Running -> Succeeded" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=info msg="node loop-test-g926s-3122930240 finished: 2023-03-29 01:18:46.006848989 +0000 UTC" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=error msg="Mark error node" error="invalid character ':' after top-level value" namespace=argo-workflows nodeName=loop-test-g926s workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=info msg="node loop-test-g926s phase Running -> Error" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=info msg="node loop-test-g926s message: invalid character ':' after top-level value" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=info msg="node loop-test-g926s finished: 2023-03-29 01:18:46.006923195 +0000 UTC" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=error msg="error in entry template execution" error="invalid character ':' after top-level value" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=info msg="Updated phase Running -> Error" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.006Z" level=info msg="Updated message  -> error in entry template execution: invalid character ':' after top-level value" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.007Z" level=info msg="Marking workflow completed" namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.007Z" level=info msg="Checking daemoned children of " namespace=argo-workflows workflow=loop-test-g926s
time="2023-03-29T01:18:46.012Z" level=info msg="cleaning up pod" action=deletePod key=argo-workflows/loop-test-g926s-1340600742-agent/deletePod
time="2023-03-29T01:18:46.022Z" level=info msg="Workflow update successful" namespace=argo-workflows phase=Error resourceVersion=742464189 workflow=loop-test-g926s
time="2023-03-29T01:18:46.081Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo-workflows/loop-test-g926s-180455059/labelPodCompleted
time="2023-03-29T01:18:46.081Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo-workflows/loop-test-g926s-407593233/labelPodCompleted
time="2023-03-29T01:18:46.081Z" level=info msg="cleaning up pod" action=labelPodCompleted key=argo-workflows/loop-test-g926s-2537052995/labelPodCompleted
```

### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l 

workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded

time="2023-03-29T01:18:41.560Z" level=info msg="S3 Save path: /tmp/argo/outputs/logs/main.log, key: argo-workflows/2023/03/29/loop-test-g926s/loop-test-g926s-180455059/main.log"
time="2023-03-29T01:18:41.560Z" level=info msg="Creating minio client using AWS SDK credentials"
time="2023-03-29T01:18:42.283Z" level=info msg="Saving file to s3" bucket=dh-ft-operations-eu-argo-workflows-artifacts endpoint=s3.amazonaws.com key=argo-workflows/2023/03/29/loop-test-g926s/loop-test-g926s-180455059/main.log path=/tmp/argo/outputs/logs/main.log
time="2023-03-29T01:18:42.337Z" level=info msg="not deleting local artifact" localArtPath=/tmp/argo/outputs/logs/main.log
time="2023-03-29T01:18:42.337Z" level=info msg="Successfully saved file: /tmp/argo/outputs/logs/main.log"
time="2023-03-29T01:18:42.337Z" level=info msg="No output parameters"
time="2023-03-29T01:18:42.337Z" level=info msg="No output artifacts"
time="2023-03-29T01:18:42.356Z" level=info msg="Create workflowtaskresults 201"
time="2023-03-29T01:18:42.357Z" level=info msg="Killing sidecars []"
time="2023-03-29T01:18:42.357Z" level=info msg="Alloc=6817 TotalAlloc=15935 Sys=23506 NumGC=5 Goroutines=12"
time="2023-03-29T01:18:41.460Z" level=info msg="S3 Save path: /tmp/argo/outputs/logs/main.log, key: argo-workflows/2023/03/29/loop-test-g926s/loop-test-g926s-2537052995/main.log"
time="2023-03-29T01:18:41.461Z" level=info msg="Creating minio client using AWS SDK credentials"
time="2023-03-29T01:18:42.500Z" level=info msg="Saving file to s3" bucket=dh-ft-operations-eu-argo-workflows-artifacts endpoint=s3.amazonaws.com key=argo-workflows/2023/03/29/loop-test-g926s/loop-test-g926s-2537052995/main.log path=/tmp/argo/outputs/logs/main.log
time="2023-03-29T01:18:42.559Z" level=info msg="not deleting local artifact" localArtPath=/tmp/argo/outputs/logs/main.log
time="2023-03-29T01:18:42.559Z" level=info msg="Successfully saved file: /tmp/argo/outputs/logs/main.log"
time="2023-03-29T01:18:42.559Z" level=info msg="No output parameters"
time="2023-03-29T01:18:42.559Z" level=info msg="No output artifacts"
time="2023-03-29T01:18:42.594Z" level=info msg="Create workflowtaskresults 201"
time="2023-03-29T01:18:42.594Z" level=info msg="Killing sidecars []"
time="2023-03-29T01:18:42.594Z" level=info msg="Alloc=6594 TotalAlloc=15535 Sys=24018 NumGC=5 Goroutines=12"
time="2023-03-29T01:18:41.507Z" level=info msg="S3 Save path: /tmp/argo/outputs/logs/main.log, key: argo-workflows/2023/03/29/loop-test-g926s/loop-test-g926s-407593233/main.log"
time="2023-03-29T01:18:41.507Z" level=info msg="Creating minio client using AWS SDK credentials"
time="2023-03-29T01:18:42.551Z" level=info msg="Saving file to s3" bucket=dh-ft-operations-eu-argo-workflows-artifacts endpoint=s3.amazonaws.com key=argo-workflows/2023/03/29/loop-test-g926s/loop-test-g926s-407593233/main.log path=/tmp/argo/outputs/logs/main.log
time="2023-03-29T01:18:42.625Z" level=info msg="not deleting local artifact" localArtPath=/tmp/argo/outputs/logs/main.log
time="2023-03-29T01:18:42.625Z" level=info msg="Successfully saved file: /tmp/argo/outputs/logs/main.log"
time="2023-03-29T01:18:42.625Z" level=info msg="No output parameters"
time="2023-03-29T01:18:42.625Z" level=info msg="No output artifacts"
time="2023-03-29T01:18:42.648Z" level=info msg="Create workflowtaskresults 201"
time="2023-03-29T01:18:42.649Z" level=info msg="Killing sidecars []"
time="2023-03-29T01:18:42.649Z" level=info msg="Alloc=6827 TotalAlloc=15741 Sys=23506 NumGC=5 Goroutines=12"
```

### PR DESCRIPTION
**Pull request by elifarley**
The current documentation doesn't provide examples on how to get the aggregate results of a loop.
This PR adds an example workflow template with some explanations.

The workflow executed successfully and there's an image to show the steps.

Closes #10775

Signed-off-by: Elifarley <elifarley@gmail.com>

### CODE REVIEW

**Comment by Joibel** [comment]
@elifarley you just need to remove the final blank line from loops.md to make this pass. Make sure you run `make docs` and commit the results after that.

File: docs/walk-through/loops.md [hunk 1]
```diff
@@ -178,3 +178,67 @@ spec:
...
+      parameters:
+      - name: aggregate-results
+        value: 'no-value'
+    script:
+      image: alpine:latest
+      command: [sh]
+      source: |
+        echo 'inputs.parameters.aggregate-results: "{{inputs.parameters.aggregate-results}}"'
+```
+
```
Comment (alexec) [inline docs/walk-through/loops.md:240]: This image needs to be added to the the `docs/assets` directory

**Review by caelan-io — APPROVED** [review]

**Review by terrytangyuan — CHANGES_REQUESTED** [review]
Please address Alex's comment above

File: docs/walk-through/loops.md [hunk 2]
```diff
@@ -176,3 +176,67 @@ spec:
...
+![image](https://user-images.githubusercontent.com/519940/228506984-554def09-2a27-4a7f-b387-8b1380889b1d.png)
```
Comment (elifarley) [inline docs/walk-through/loops.md:239]:
```suggestion
![image](../../aggregate-result-of-all-iterations-of-a-loop.png)
```

File: docs/walk-through/loops.md [hunk 3]
```diff
@@ -176,3 +176,67 @@ spec:
...
+      - name: aggregate-results
+        value: 'no-value'
+    script:
+      image: alpine:latest
+      command: [sh]
+      source: |
+        echo 'inputs.parameters.aggregate-results: "{{inputs.parameters.aggregate-results}}"'
+```
+
+![image](../../aggregate-result-of-all-iterations-of-a-loop.png)
```
Comment (elifarley) [inline docs/walk-through/loops.md:239]:
```suggestion
![image](../aggregate-result-of-all-iterations-of-a-loop.png)
```

File: docs/walk-through/loops.md [hunk 4]
```diff
@@ -176,3 +176,67 @@ spec:
...
+      - name: aggregate-results
+        value: 'no-value'
+    script:
+      image: alpine:latest
+      command: [sh]
+      source: |
+        echo 'inputs.parameters.aggregate-results: "{{inputs.parameters.aggregate-results}}"'
+```
+
+![image](../aggregate-result-of-all-iterations-of-a-loop.png)
```
Comment (elifarley) [inline docs/walk-through/loops.md:239]:
```suggestion
![image](../assets/aggregate-result-of-all-iterations-of-a-loop.png)
```

File: docs/walk-through/loops.md (hunk 1 above)
Comment (elifarley) [reply docs/walk-through/loops.md:238]: @alexec @Joibel @terrytangyuan Done.

**Review by terrytangyuan — APPROVED** [review]

**Comment by terrytangyuan** [comment]
Please sign DCO

**Pull request merged**